File: LeetCode/Medium/0763-partition-labels/0763-partition-labels.go
func partitionLabels(s string) []int {
    idxs := make([]int, 26)

    for i := 0; i < len(s); i++ { idxs[s[i]-97] = i }

    ret := []int{}
    end, size := 0, 0

    for i := 0; i < len(s); i++ {
        size++
        end = max(end, idxs[s[i]-97])

        if i == end { 
            ret = append(ret, size)
            size = 0
        }
    }

    return ret
}

func max(a, b int) int {
    if a > b { return a }
    return b
}

File: LeetCode/Medium/0763-partition-labels/0763-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        n = len(s)
        idxs = [[] for i in range(26)]

        for i in range(n):
            idxs[ord(s[i])-97].append(i)
        
        start = idxs[ord(s[0])-97][0]
        end = idxs[ord(s[0])-97][-1]
        ret = []

        for i, v in enumerate(s):
            idx = ord(v) - 97
            if end < idxs[idx][0]:
                ret.append(end - start + 1)
                start = idxs[idx][0]
                end = idxs[idx][-1]
            elif end >= idxs[idx][-1]:
                continue
            else:
                end = idxs[idx][-1]
        
        ret.append(end-start+1)

        return ret
            
        
            